File: Assignment35_3.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : Difference
// Description   : Calculate difference between frequency of small and capital characters
// Input         : String
// Output        : Integer (Difference)
// Author        : Raj Samir Jadhav
// Date          : 03/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    public int Difference(String str)          // Function
    {
        int iSmall = 0, iCapital = 0;

        for(int i = 0; i < str.length(); i++)
        {
            char ch = str.charAt(i);

            if(ch >= 'a' && ch <= 'z')        // Small letter
            {
                iSmall++;
            }
            else if(ch >= 'A' && ch <= 'Z')   // Capital letter
            {
                iCapital++;
            }
        }
        return iSmall - iCapital;             // Return difference
    }
}

class Assignment35_3
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        int iRet = pobj.Difference(input);

        System.out.println("Difference is : " + iRet);

        sobj.close();                       // Close scanner to avoid warning
    }
}

File: Assignment35_3.c
#include <stdio.h>
///////////////////////////////////////////////////////////////////////////////
// Function Name : Difference
// Description   : Calculate difference between frequency of small and capital characters
// Input         : String
// Output        : Integer (Difference)
// Author        : Raj Samir Jadhav
// Date          : 03/12/2025
///////////////////////////////////////////////////////////////////////////////
int Difference(char *str)
{
    int iSmall = 0, iCapital = 0;

    while(*str != '\0')                         // Traverse string
    {
        if((*str >= 'a') && (*str <= 'z'))      // Count small letters
        {
            iSmall++;
        }
        else if((*str >= 'A') && (*str <= 'Z'))  // Count capital letters
        {
            iCapital++;
        }
        str++;                                    // Move to next character
    }

    return (iSmall - iCapital);                   // Return difference
}

int main()
{
    char arr[20];
    int iRet = 0;

    printf("Enter string : ");
    scanf("%[^'\n']s",arr);

    iRet = Difference(arr);

    printf("Difference is : %d\n", iRet);

    return 0;
}
